### ISSUE TITLE
Subtraction overflow in matrix-sdk-crypto/machine.rs

### ISSUE THREAD

**Issue by DevinR528** [matrix-org/matrix-rust-sdk#58]
`should_upload_keys` subtraction overflowed on me.
```rust
match &self.uploaded_signed_key_count {
    Some(count) => {
        let max_keys = self.account.max_one_time_keys().await as u64;
        let key_count = (max_keys / 2) - count.load(Ordering::Relaxed);
            key_count > 0 // should this be an i64 instead of u64?
        }
    None => false,
}
```

**Comment by poljar** [comment]
I think that means that you somehow have more keys uploaded than the sdk usually keeps one the server.

No, I don't think we need to make this an i64, just make sure to return false if the current uploaded count is bigger than `max_keys / 2`.

**Comment by DevinR528** [comment]
Closed by #59
